src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
> the pattern "certain classes of code must be in a separate function, even if it is very short and integral with the surrounding code" I have never heard of.

The API functions and `foo_event` handlers are somewhat analogous. Regularity is really helpful. And given that these functions are part of a "class" as you noted, formalizing the name under a namespace seems natural.